src/pe/data_directories.rs

```diff
@@ -1,20 +1,111 @@
 use crate::error;
-use scroll::{Pread, Pwrite, SizeWith};
+use core::{ops::Deref, iter::Flatten};
+use scroll::{ctx, Pread, Pwrite, SizeWith};
 
 #[repr(C)]
 #[derive(Debug, PartialEq, Copy, Clone, Default, Pread, Pwrite, SizeWith)]
-pub struct DataDirectory {
+pub struct DataDirectoryInner {
     pub virtual_address: u32,
     pub size: u32,
 }
+#[derive(Debug, PartialEq, Copy, Clone, Default)]
+pub struct DataDirectory {
+    pub inner: DataDirectoryInner,
+    pub(crate) offset: usize,
+}
 
 pub const SIZEOF_DATA_DIRECTORY: usize = 8;
 const NUM_DATA_DIRECTORIES: usize = 16;
 
 impl DataDirectory {
     pub fn parse(bytes: &[u8], offset: &mut usize) -> error::Result<Self> {
-        let dd = bytes.gread_with(offset, scroll::LE)?;
-        Ok(dd)
+        let inner = bytes.gread_with(offset, scroll::LE)?;
+        Ok(DataDirectory {
+            inner,
+            offset: *offset,
```

Comment:
Could I have your feedback on this @m4b before I go further?
I am not sure how to compute `offset` except by getting at parsing time.